activesupport/CHANGELOG.md

```diff
@@ -1,3 +1,9 @@
+*   Deprecated initializing a `ActiveSupport::Cache::MemCacheStore` with an instance of `Dalli::Client`.
+
+    Previously providing an instance of `Dalli::Client` to `ActiveSupport::Cache::MemCacheStore` was allowed, but the difference in behavior between `Dalli::Client` and `ActiveSupport::Cache::MemCacheStore` introduces inconsistent behavior. For example reading multiple keys, when the client instance has a custom namespace would not work as expected.
```

Comment:
```suggestion
    Deprecate the undocumented option of providing an already-initialized instance of `Dalli::Client` to `ActiveSupport::Cache::MemCacheStore`. Such clients could be configured with unrecognized options, which could lead to unexpected behavior. Instead, provide addresses as documented.
```

I think it's clearer to use a more direct voice and conclude with an action item.

I tightened up the explanation in my second sentence, but I would also support removing that whole sentence. It's maybe too wordy for the ActiveSupport changelog. If people really care why it's being deprecated they can look up this PR :)

By the way, this option was added in Rails 3.0 (e2ed1a1ca, af44b07649), and then Rails 4.1 (98458eea42) changed it slightly while removing the documentation for it. So this has been an undocumented option since 2014.